### ISSUE TITLE
anvil_repair does not work

### ISSUE THREAD

**Issue by evilsaloon** [PluginBugs/Issues-ItemsAdder#2819]
### Terms

- [X] I'm using the very latest version of ItemsAdder and its dependencies.
- [X] I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
- [X] I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
- [X] I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
- [X] I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.

### Discord tag (optional)

evilsaloon

### What happened?

Anvil repair with vanilla and custom IA items does not work with 1.20 update.

### Steps to reproduce the issue

1. Add item .yml
2. /iareload
3. Repair does not work in anvil (see screenshot)

### Server version

Current: git-Purpur-2001 (MC: 1.20.1)*
Previous: git-Purpur-1996 (MC: 1.20.1)
* You are 3 version(s) behind
Download the new version at: https://purpurmc.org/downloads

### ItemsAdder Version

ItemsAdder version 3.5.0b

### ProtocolLib Version

ProtocolLib version 5.1.0-SNAPSHOT-654

### LoneLibs Version

LoneLibs version 1.0.25

### FULL server log

https://pastebin.com/7H4C8uu0

### Error (optional)

```shell
No error in console.
```

### Problematic items yml configuration file (optional)

```shell
info:
  namespace: testserv
items:
  testium_sword:
    nbt: "{display:{Name:'{\"italic\":false,\"translate\":\"item.testserv.testium_sword\"}'}}"
    permission: testserv
    durability:
       max_custom_durability: 2032
    attribute_modifiers:
      mainhand:
        attackDamage: 8
        attackSpeed: 2.6
      offhand:
        attackDamage: 8
        attackSpeed: 2.6
    resource:
      material: DIAMOND_SWORD
      generate: true
      textures:
        - testium/testium_sword.png
anvil_repair:
  testium_sword:
    permission: testserv
    ingredient: testserv:testium_ingot
    item: testserv:testium_sword
```

### Screenshots/Videos (you can drag and drop files or paste links)

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/80675840/02a2ed72-9bf8-4070-87db-8652697f4eac)

**Comment by yorik100** [comment]
> ### Terms
> 
>     * [x]  I'm using the very latest version of ItemsAdder and its dependencies.
> 
>     * [x]  I am sure this is a bug and it is not caused by a misconfiguration or by another plugin.
> 
>     * [x]  I already searched on this [Github page](https://github.com/PluginBugs/Issues-ItemsAdder/issues) to check if the same issue was already reported.
> 
>     * [x]  I already searched on the [plugin wiki](https://itemsadder.devs.beer/) to know if a solution is already known.
> 
>     * [x]  I already searched on the [forums](https://forum.devs.beer/) to check if anyone already has a solution for this.
> 
> 
> ### Discord tag (optional)
> 
> evilsaloon
> ### What happened?
> 
> Anvil repair with vanilla and custom IA items does not work with 1.20 update.
> ### Steps to reproduce the issue
> 
>     1. Add item .yml
> 
>     2. /iareload
> 
>     3. Repair does not work in anvil (see screenshot)
> 
> 
> ### Server version
> 
> Current: git-Purpur-2001 (MC: 1.20.1)* Previous: git-Purpur-1996 (MC: 1.20.1)
> 
>     * You are 3 version(s) behind
>       Download the new version at: https://purpurmc.org/downloads
> 
> 
> ### ItemsAdder Version
> 
> ItemsAdder version 3.5.0b
> ### ProtocolLib Version
> 
> ProtocolLib version 5.1.0-SNAPSHOT-654
> ### LoneLibs Version
> 
> LoneLibs version 1.0.25
> ### FULL server log
> 
> https://pastebin.com/7H4C8uu0
> ### Error (optional)
> 
> ```shell
> No error in console.
> ```
> 
> ### Problematic items yml configuration file (optional)
> 
> ```shell
> info:
>   namespace: testserv
> items:
>   testium_sword:
>     nbt: "{display:{Name:'{\"italic\":false,\"translate\":\"item.testserv.testium_sword\"}'}}"
>     permission: testserv
>     durability:
>        max_custom_durability: 2032
>     attribute_modifiers:
>       mainhand:
>         attackDamage: 8
>         attackSpeed: 2.6
>       offhand:
>         attackDamage: 8
>         attackSpeed: 2.6
>     resource:
>       material: DIAMOND_SWORD
>       generate: true
>       textures:
>         - testium/testium_sword.png
> anvil_repair:
>   testium_sword:
>     permission: testserv
>     ingredient: testserv:testium_ingot
>     item: testserv:testium_sword
> ```
> 
> ### Other files, you can drag and drop them here to upload. (optional)
> 
> _No response_
> ### Screenshots/Videos (you can drag and drop files or paste links)
> 
> ![image](https://user-images.githubusercontent.com/80675840/248911467-02a2ed72-9bf8-4070-87db-8652697f4eac.png)

Works fine for me, you can try using the iageneric -> iasurvival as an example https://github.com/ItemsAdder/DefaultPack/releases

**Comment by LoneDev6** [comment]
Works fine for me.
There is probably a misconfiguration in your configs.

![image](https://github.com/PluginBugs/Issues-ItemsAdder/assets/27242001/a9d28ca9-dea2-4c07-a52d-caefcf2a6c12)

[Item](https://github.com/ItemsAdder/DefaultPack/blob/82a3d2fa786a8901f86c7fc7ea580bbc58fb2ce8/contents/iasurvival/configs/swords/items/ruby_sword.yml#L4)
[Recipe](https://github.com/ItemsAdder/DefaultPack/blob/82a3d2fa786a8901f86c7fc7ea580bbc58fb2ce8/contents/iasurvival/configs/swords/recipes_anvil.yml#L15)

**Comment by evilsaloon** [comment]
Yep, it does work. I was going off the wiki (https://itemsadder.devs.beer/plugin-usage/adding-content/recipes/anvil-repair) and thought you didn't need to have a "recipes:" layer before "anvil_repair:". My bad, sorry.

Here's a working config in case someone has the same issue:

recipes:
  anvil_repair:
    emerald_sword:
      enabled: true
      permission: iasurvival.swords.emerald_sword
      ingredient: EMERALD_BLOCK
      item: iasurvival:emerald_sword
